superset/connectors/sqla/models.py

```diff
@@ -749,7 +752,15 @@ def get_sqla_query(  # sqla
                     "and is required by this type of chart"
                 )
             )
-        if not groupby and not metrics and not columns:
+        if (
+            IS_SIP_38
+            and not metrics
+            and not columns
+            or not IS_SIP_38
```

Comment:
See ^^. Also if any of these conditions are the same a utility function may be helpful.